alfasim_sdk/hook_specs.py

```diff
@@ -1333,6 +1333,52 @@ def calculate_entrained_liquid_fraction(
     """
 
 
+def update_plugins_internal_deposit_layer(ctx: "void*", deposition_layer_thickness: "void*",
+                                             n_control_volumes: "int") -> "int":
+    """
+    **c++ signature** : ``HOOK_UPDATE_PLUGINS_INTERNAL_DEPOSIT_LAYER(void* ctx, void* deposition_layer_thickness,
+    int n_control_volumes)``
+
+    Internal simulator hook to evaluate the thickness of the deposited layer at the inside of the pipeline walls.
+    This is called at the beginning of accounting the diameter reduction.
+
+    The plugin is supposed to change the given ``deposition_layer_thickness`` array pointer. Its values are contiguous
```

Comment:
Hmmm is not entirely clear if the plugin should add/reduce from the existing values, or set a value blindly. Also will does play out when multiple plugins implement this hook?

If multiple plugins should just "add" their thickness to each control volume, won't this be a problem in case the plugin needs to track the amount of deposited layer from the previous time step? Is the plugin supposed to track this using their own variables, initialized in `initialize`?

Reply:
The plugin should set the value calculated for the thickness at the time-step and not set the difference of thickness between two time-steps. 
So, as an example, the wax plugin calculates the first wax deposit layer thickness, saves this value as a plugin secondary variable (called `wax_thickness`),  calculates the variation of the thickness between 2 time-steps and it evaluates the new thickness by summing the difference with `wax_thickness_old`.

Reply:
About multiple plugins are not yet supported for all hooks, but the way is done is analogous to other hooks (such as `calculate_mass_source_term`) and it is ready for when the task for multiple plugins be done.